src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1547,5 +1547,24 @@ internal bool Equals(TypeSymbolWithAnnotations other)
         {
             throw ExceptionUtilities.Unreachable;
         }
+
+        public static bool Equals(TypeSymbol left, TypeSymbol right, TypeCompareKind comparison)
+        {
+            if (comparison == TypeCompareKind.ConsiderEverything && ((object)right == null))
+            {
+                return (object)left == (object)null;
+            }
+
+            return (comparison == TypeCompareKind.ConsiderEverything && (object)left == (object)right)
+                || right.Equals(left, comparison);
```

Comment:
>right.Equals(left, comparison) [](start = 19, length = 30)

It feels strange that we are reversing the sides, also the ```comparison``` checks are redundant.  Consider the following alternative implementation for this function:
```
if ((object)left == null)
{
    return  (object)right == null;
}

return left.Equals(right, comparison);
```